internal/charmstore/store_test.go

```diff
@@ -2030,6 +2030,74 @@ func (s *StoreSuite) TestPromulgate(c *gc.C) {
 	}
 }
 
+func (s *StoreSuite) TestSetPromulgatedUpdateSearch(c *gc.C) {
```

Comment:
Some commentary indicating what's being tested in each section of this function would be useful, I think.